src/main/java/com/snowflake/kafka/connector/internal/streaming/telemetry/SnowflakeTelemetryChannelStatus.java

```diff
@@ -18,23 +18,32 @@
 package com.snowflake.kafka.connector.internal.streaming.telemetry;
 
 import static com.snowflake.kafka.connector.internal.metrics.MetricsUtil.constructMetricName;
+import static com.snowflake.kafka.connector.internal.streaming.TopicPartitionChannel.NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE;
 
 import com.codahale.metrics.Gauge;
 import com.codahale.metrics.MetricRegistry;
-import com.snowflake.kafka.connector.internal.KCLogger;
+import com.google.common.annotations.VisibleForTesting;
 import com.snowflake.kafka.connector.internal.metrics.MetricsJmxReporter;
 import com.snowflake.kafka.connector.internal.metrics.MetricsUtil;
+import com.snowflake.kafka.connector.internal.telemetry.SnowflakeTelemetryBasicInfo;
+import com.snowflake.kafka.connector.internal.telemetry.SnowflakeTelemetryService;
+import com.snowflake.kafka.connector.internal.telemetry.TelemetryConstants;
 import java.util.concurrent.atomic.AtomicLong;
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.databind.node.ObjectNode;
 
-public class SnowflakeTelemetryChannelStatus {
-  private static final KCLogger LOGGER =
-      new KCLogger(SnowflakeTelemetryChannelStatus.class.toString());
-
+/**
+ * Extension of {@link SnowflakeTelemetryBasicInfo} class used to send data to snowflake when the
+ * TopicPartitionChannel closes. Also creates and registers various metrics with JMX
+ *
+ * <p>Most of the data sent to Snowflake is an aggregated data.
+ */
+public class SnowflakeTelemetryChannelStatus extends SnowflakeTelemetryBasicInfo {
   public static final long NUM_METRICS = 3; // update when new metrics are added
 
   // channel properties
   private final String channelName;
   private final MetricsJmxReporter metricsJmxReporter;
+  private final AtomicLong startTime;
```

Comment:
why we need AtomicLong here?

Reply:
I'll change it to a normal final long.

startTime represents when the SnowflakeTelemetryChannelStatus object started tracking data. it gets updated to the current time every time we send telemetry data, these were behaviors I took over from the corresponding Snowpipe object.

Currently since we only send telemetry data once, it can be a final long, however in the future if we want to send multiple SnowflakeTelemetryChannelStatus's, an AtomicLong may make more sense for threading.